programs/drift/src/state/perp_market.rs

```diff
@@ -1689,3 +1696,152 @@ impl AMM {
         }
     }
 }
+
+pub const AMM_POSITIONS_CACHE: &str = "amm_positions_cache";
+
+#[account]
+#[derive(Debug)]
+#[repr(C)]
+pub struct AmmCache {
+    pub bump: u8,
+    _padding: [u8; 3],
+    pub cache: Vec<CacheInfo>,
+}
+
+#[zero_copy]
+#[derive(AnchorSerialize, AnchorDeserialize, Debug)]
+#[repr(C)]
+pub struct CacheInfo {
+    pub last_fee_pool_token_amount: u128,
+    pub last_net_pnl_pool_token_amount: i128,
+    /// BASE PRECISION
+    pub position: i64,
+    pub slot: u64,
+    pub max_confidence_interval_multiplier: u64,
+    pub last_oracle_price_twap: i64,
+    pub last_settle_amount: u64,
+    pub last_settle_ts: i64,
+    pub quote_owed_from_lp: i64,
+    pub oracle_price: i64,
+    pub oracle_confidence: u64,
+    pub oracle_delay: i64,
+    pub oracle_slot: u64,
+    pub oracle_source: u8,
+    _padding: [u8; 7],
+    pub oracle: Pubkey,
```

Comment:
put this at top

Reply:
can we hit any weird race condition if we update oracle on perp market and this isn't updated?